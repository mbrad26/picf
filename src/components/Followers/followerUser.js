import React, { lazy } from 'react';
import { Image } from 'react-bootstrap';

const Username = lazy(() => import('../Following/username'));

const FollowerUser = ({ user }) => {
  return (
    <>
      <Image loading='eager' id='avatar' src={user.avatarUrl} roundedCircle />
      <Username username={user.username} />
    </>
  );
};

export default FollowerUser;