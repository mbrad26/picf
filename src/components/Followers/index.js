import React, { useEffect, lazy, Suspense } from 'react';
import { Link } from 'react-router-dom';
import { Nav } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

import { doFollowStatusRequest } from '../../redux/actions/user';

const FollowerUser = lazy(() => import('./followerUser'));

const Followers = () => {
  const dispatch = useDispatch();
  const { followers } = useSelector(state => state.userState);

  useEffect(() => {
    dispatch(doFollowStatusRequest());
  }, [dispatch]);

  return (
    <>
      <p className='titles'>Followers</p>

      <Suspense fallback={<p>Loading...</p>}>
        {followers &&
          followers.map(user => 
            <div key={user.uid}>
              <div className='sidebar-avatars'>
                <Nav.Link as={Link} to={`/home/followers/${user.uid}`}>
                  <FollowerUser user={user} />
                </Nav.Link>
              </div>
            </div>
          )
        }
      </Suspense>
    </>
  );
};

export default Followers;
