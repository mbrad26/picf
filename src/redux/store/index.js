import { createLogger } from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import { createStore, applyMiddleware } from  'redux';

import rootReducer from '../reducers';
import rootSaga from '../sagas';

const logger = createLogger();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  {},
  applyMiddleware(saga, logger)
);

saga.run(rootSaga);

export default store;