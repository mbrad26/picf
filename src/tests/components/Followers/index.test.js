import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import store from '../../../redux/store';
import Followers from  '../../../components/Followers/index';

describe('Account', () => {
  it('renders snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Followers />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});