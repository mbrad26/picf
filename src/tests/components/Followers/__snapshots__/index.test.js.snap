// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account renders snapshot 1`] = `
<p
  class="titles"
>
  Followers
</p>
`;
